lib/ruby_lsp/requests/request.rb

```diff
@@ -0,0 +1,36 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    # :nodoc:
+    class Request
+      extend T::Sig
+      extend T::Generic
+
+      abstract!
+
+      sig { abstract.returns(T.anything) }
+      def response; end
```

Comment:
What do you think about maintaining the requests' main method as `run`? That will reduce the amount of changes, but I also think it's a better name then `response` in this case.

Reply:
I extracted the standardisation changes to #1281. But I prefer `response` over `run`.

Reply:
I still find it odd to name the request's main method as `response`. It hints that requests do not control their own flow of execution, like listeners.

I really think `run` or `perform` would be a better fit. @andyw8 do you have thoughts on the naming?

Reply:
Yeah, yeah. Let's do it in the next PR, no need to do it in this one.

My opinion is that having `response` for listeners and `run/perform` for requests makes the naming a bit clearer. It also avoids confusion since many listeners and requests are named the same.

Please, do move forward with this PR and we can change the name of the request methods in the next.

Reply:
I've opened https://github.com/Shopify/ruby-lsp/pull/1297 for renaming.